lib/screens/all_tags_screen.dart

```diff
@@ -76,10 +76,63 @@ class _AllTagsScreen extends State<AllTagsScreen> {
                   ),
                   Wrap(
                     children: _assignedTags
-                        .map((e) => Padding(
-                              padding: const EdgeInsets.all(8.0),
-                              child: Tag(tagName: e),
-                            ))
+                        .map((e) =>
+                            Container(
+                                  padding: const EdgeInsets.all(8.0),
+                                  child: Row(
+                                    children: [
+                                      Tag(tagName: e),
+                                      Container(
+                                        padding: const EdgeInsets.only(left: 8.0),
+                                        child: CircleAvatar(
+                                          backgroundColor:
+                                          const Color.fromRGBO(122, 83, 217, 0.9),
+                                          child: IconButton(
+                                            onPressed: () async {
+                                              String tagInQuestion = e;
+                                              myUser.tags.remove(tagInQuestion);
+                                              FirebaseFirestore.instance
+                                                  .collection("users")
+                                                  .doc(userid)
+                                                  .set(myUser.toJson());
+                                              var snapshot = await FirebaseFirestore
+                                                  .instance
+                                                  .collection("groups")
+                                                  .get();
+                                              snapshot.docs.forEach(
```

Comment:
This part of the code has the following purpose: When a new tag is assigned to a user, we iterate over all the groups and check if the user can be added to a new group after the assignment of the tag. As this part is not relevant to unassigning a tag, please remove this.